x-pack/plugins/maps/public/routing/routes/maps_app/maps_app_view.js

```diff
@@ -356,22 +364,20 @@ export class MapsAppView extends React.Component {
     );
   }
 
-  render() {
-    const { filters, isFullScreen } = this.props;
+  _addFilter = (newFilters) => {
+    newFilters.forEach((filter) => {
+      filter.$state = { store: esFilters.FilterStateStore.APP_STATE };
+    });
+    this._onFiltersChange([...this.props.filters, ...newFilters]);
+  };
 
+  render() {
     return this.state.initialized ? (
-      <div id="maps-plugin" className={isFullScreen ? 'mapFullScreen' : ''}>
+      <div id="maps-plugin" className={this.props.isFullScreen ? 'mapFullScreen' : ''}>
         {this._renderTopNav()}
         <h1 className="euiScreenReaderOnly">{`screenTitle placeholder`}</h1>
         <div id="react-maps-root">
-          <MapContainer
-            addFilters={(newFilters) => {
-              newFilters.forEach((filter) => {
-                filter.$state = { store: esFilters.FilterStateStore.APP_STATE };
-              });
-              this._onFiltersChange([...filters, ...newFilters]);
-            }}
-          />
+          <MapContainer addFilters={this._addFilter} />
```

Comment:
agreed that extracting this adds clarity, but isn't this functionally equivalent?  The lambda makes sure `this` get bounds to the parent scope from `this.render()` call, which should be the React-component. What am I missing?

Reply:
They are functionally equivalent. The change is that the original code created a new function on every render. The new code uses the same function reference on every render. This is important because down stream `DrawControl` `componentDidUpdate` will get called every time any of the props change. This means that even though the function is the same, `componentDidUpdate` gets called because the function reference has changed.